engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,10 +492,21 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog()) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
+      }
     }
   }
 
+  private boolean shouldLog()
+  {
+    if ((lastLog == -1) || (totalTupleCount - lastLog) >= ERROR_LOG_GAP) {
```

Comment:
assuming that errors don't occur frequently, logGap will be 0 most of the time and there will be no decrement.